src/number_theory/cyclotomic/discriminant.lean

```diff
@@ -134,6 +134,73 @@ begin
   { apply_instance }
 end
 
+/-- If `p` is a prime and `is_cyclotomic_extension {p ^ (k + 1)} K L`, then the discriminant of
+`hζ.power_basis K` is `(-1) ^ ((p ^ (k + 1).totient) / 2) * p ^ (p ^ k * ((p - 1) * (k + 1) - 1))`
+if `irreducible (cyclotomic (p ^ (k + 1)) K))`, `irreducible (cyclotomic p K)`. Beware that the
+formula uses `1 / 2 = 0` and it is used only to have a uniform result. See also
+`discr_prime_pow_eq_unit_mul_pow`. -/
+lemma discr_prime_pow [is_cyclotomic_extension {p ^ (k + 1)} K L] [hp : fact (p : ℕ).prime]
+  [ne_zero ((p : ℕ) : K)] (hζ : is_primitive_root ζ ↑(p ^ (k + 1)))
+  (hirr : irreducible (cyclotomic (↑(p ^ (k + 1)) : ℕ) K))
+  (hirr₁ : irreducible (cyclotomic (p : ℕ) K)) :
+  discr K (hζ.power_basis K).basis =
+  (-1) ^ (((p ^ (k + 1) : ℕ).totient) / 2) * p ^ ((p : ℕ) ^ k * ((p - 1) * (k + 1) - 1)) :=
```

Comment:
Actually if you do replace `k+1` by `k` then `(-1) ^ (((p ^ k : ℕ).totient) / 2) * p ^ ((p : ℕ) ^ (k - 1) * ((p - 1) * k - 1))` still work when `k = 0` because `0 - 1 = 0` ...

Reply:
Nice catch!